backend/src/db.js

```diff
@@ -26,3 +26,5 @@ export default async function connectDB(dbName) {
     process.exit(1);
   }
 }
+
+export { mongoose as db };
```

Comment:
I'm not sure if I understand why this line is needed? It doesn't seem like anything happens with mongoose, and other classes could just import mongoose directly rather than through this module. Could you please explain what the intent behind this is?

Reply:
Think of it like a Singleton. If we export the same instance of mongoose we can run operations against it anywhere in the code base. Mongoose helps establish a connection to a Mongo database and close the connection when we don't need it. In the index.test.ts file, after running the test, Jest said that it detected an open handle and it didn't exit, this is because the MongoDB connection was still open. The Mongoose instance I exported from db.js lets us close that connection and properly disconnect from the DB.

Reply:
I see so you're trying to cache the mongoose connection right? I'm not sure if it's needed given that when you call mongoose.connect(), mongoose creates a default connection that you access with mongoose.connection (https://mongoosejs.com/docs/connections.html#multiple_connections). What was the exact error you were getting? The only kind of connection error I get is when the Assistant.js test suite runs, and I kinda expect that because of things are setup on my computer right now.

I see that one way of caching a mongoose connection the docs recommends is to store the connection promise in a variable and check it before calling mongoose.connect() again (https://mongoosejs.com/docs/lambda.html). Whether or not you want to do it this way seems to depend on how you deploy the app. I'd think about this later in a separate issue.

Reply:
So in other words mongoose itself is a singleton. Importing mongoose from mongoose as opposed to from `db.js` should close the connection. And I tested it, and it does! The issue I was facing was that the test wasn't exiting cleanly because of an [open handle](https://jestjs.io/docs/cli#--detectopenhandles) due mongoose not being explicitly closed. But importing mongoose itself made it work! I'll revert this file.